src/org/zaproxy/zap/extension/exportreport/ExtensionExportReport.java

```diff
@@ -0,0 +1,290 @@
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ * This file is based on the Paros code file ReportLastScan.java
+ */
+package org.zaproxy.zap.extension.exportreport;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+
+import org.zaproxy.zap.extension.exportreport.Export.ExportReport;
+import org.zaproxy.zap.extension.exportreport.Utility.SharedFunctions;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+/*
+ * AUTHOR		:	GORAN SARENKAPA - JordanGS
+ * SPONSOR		:	RYERSON UNIVERSITY
+ */
+
+public class ExtensionExportReport extends ExtensionAdaptor
+{
+
+	private static final String			NAME				= "ExtensionExportReport";
+	private static final String			FONT				= "Arial";
+	
+	private ZapMenuItem					menuExportReport	= null;
+	private FrameExportReport			frameER				= null;
+
+	protected final Map<String, String>	alertTypeRisk		= new HashMap<>();
+
+	/* Nav Panel creation. */
+	private PanelSource					cardSource			= null;
+	private PanelAlertRisk				cardAlertRisk		= null;
+	private PanelAlertDetails			cardAlertDetails	= null;
+
+	/* Field Limit Constants */
+	private final int					textfieldLimit		= 30;
+	private final int					textareaLimit		= 600;
+
+	/* Relational CONSTANT for export types. */
+	private ArrayList<String>			alertSeverity		= new ArrayList<String>();
+	private ArrayList<String>			alertDetails		= new ArrayList<String>();
+	private ArrayList<String>			alertAdditional		= new ArrayList<String>();
+	private int							maxList				= 0;
+	
+	public ExtensionExportReport()
+	{
+		super();
+		initialize();
+	}
+
+	public ExtensionExportReport(String name)
+	{
+		super(name);
+	}
+
+	private void initialize()
+	{
+		this.setName(NAME);
+		SharedFunctions.fontExists(FONT);		
+	}
+
+	@Override
+	public void hook(ExtensionHook extensionHook)
+	{
+		super.hook(extensionHook);
+		if (getView() != null)
+		{
+			extensionHook.getHookMenu().addReportMenuItem(getMenuExportReport());
+		}
+	}
+
+	@Override
+	public boolean canUnload()
+	{
+		return true;
+	}
+
+	@Override
+	public void unload()
+	{
+		super.unload();
+		if (frameER != null)
+		{
+			frameER.dispose();
+			frameER = null;
+		}
+	}
+
+	private ZapMenuItem getMenuExportReport()
+	{
+		if (menuExportReport == null)
+		{
+			menuExportReport = new ZapMenuItem("exportreport.menu.export.label");
```

Comment:
Missed function call here, and the following string probably isn't necessary. Perhaps:

`ZapMenuItem(Constant.messages.getString("exportreport.menu.export.label"))`